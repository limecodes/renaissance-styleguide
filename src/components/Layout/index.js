import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import Header from '../Header'
import Sidebar from '../Sidebar'

const Container = styled.main`
  display: flex;
`

const PageWrapper = styled.section`
  background-color: transparent;
  width: 100%;
  height: 100vh;
`

const Layout = (props) => {
  const { withHeader } = props

  return (
    <Container>
      <Sidebar />
      <PageWrapper>{withHeader && <Header />}</PageWrapper>
    </Container>
  )
}

Layout.propTypes = {
  withHeader: PropTypes.bool,
}

Layout.defaultProps = {
  withHeader: true,
}

export default Layout
