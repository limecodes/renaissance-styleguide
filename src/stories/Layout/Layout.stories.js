import React from 'react'

import Layout from '../../components/Layout'

export default {
  title: 'Layout/Main',
  component: Layout,
}

const Template = (args) => <Layout {...args} />

export const Main = Template.bind({})
Main.args = {}
