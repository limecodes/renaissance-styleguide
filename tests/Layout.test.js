import React from 'react'
import renderer from 'react-test-renderer'
import { library } from '@fortawesome/fontawesome-svg-core'
import { faBars } from '@fortawesome/free-solid-svg-icons'

import Layout from '../src/components/Layout'

library.add(faBars)

test('Layout Renders Correctly', () => {
  const tree = renderer.create(<Layout />)

  expect(tree).toMatchSnapshot()
})
