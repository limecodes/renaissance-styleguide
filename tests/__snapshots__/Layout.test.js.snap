// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Renders Correctly 1`] = `
<main
  className="sc-gKAblj fvRAFo"
>
  <section
    className="sc-jSFkmK ljJbPe"
  >
    Sidebar
  </section>
  <section
    className="sc-iCoHVE ismwmQ"
  >
    <div
      className="sc-bdnylx gunsZc"
    >
      <ul
        className="sc-gtssRu hoQdqF"
      >
        <li
          className="sc-dlnjPT jKgXLC"
        >
          <button
            className="sc-eCApGN dPGIcb"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-bars fa-w-14 "
              data-icon="bars"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </li>
        <li
          className="sc-dlnjPT jKgXLC"
        >
          <a
            className="sc-hKFyIo gVPA-dU"
            href="/home"
          >
            Home
          </a>
        </li>
        <li
          className="sc-dlnjPT jKgXLC"
        >
          <a
            className="sc-hKFyIo gVPA-dU"
            href="/patients"
          >
            Patients
          </a>
        </li>
      </ul>
    </div>
  </section>
</main>
`;
